### PR TITLE
Fix/sync entity schemas

### PR DESCRIPTION
**Pull request by GGabriele**
Right now JSON entity schemas are missing most default values. Let's just add them so that these defaults can be exposed in the schemas endpoints

### CODE REVIEW

File: internal/resource/service.go [hunk 1]
```diff
@@ -15,11 +15,12 @@ import (
...
 const (
-	defaultTimeout = 60000
-	defaultRetries = 5
-	defaultPort    = 80
-	maxRetries     = 32767
-	maxVerifyDepth = 64
+	defaultRetries  = 5
+	defaultPort     = 80
+	maxPort         = 65535
+	maxRetries      = 32767
```
Comment (tjasko) [inline internal/resource/service.go:21]: nit:
```suggestion
	maxRetries      = math.MaxInt16
```

File: internal/resource/service.go [hunk 2]
```diff
@@ -15,11 +15,12 @@ import (
...
 
 const (
-	defaultTimeout = 60000
-	defaultRetries = 5
-	defaultPort    = 80
-	maxRetries     = 32767
-	maxVerifyDepth = 64
+	defaultRetries  = 5
+	defaultPort     = 80
+	maxPort         = 65535
```
Comment (tjasko) [inline internal/resource/service.go:20]: nit:
```suggestion
	maxPort         = math.MaxUint16
```

File: internal/resource/upstream.go [hunk 3]
```diff
@@ -56,30 +67,68 @@ var (
...
+		"tcp_failures":  0,
+		"http_statuses": defaultPassiveUnhealtyHTTPStatuses,
+		"timeouts":      0,
+	}
+	defaultActive = map[string]interface{}{
+		"concurrency":              defaultConcurrency,
+		"healthy":                  defaultActiveHealthy,
+		"http_path":                "/",
+		"https_verify_certificate": true,
+		"type":                     "http",
```
Comment (tjasko) [inline internal/resource/upstream.go:118]: nit: 
```suggestion
		"type":                     typedefs.ProtocolHTTP,
```

File: internal/resource/upstream.go [hunk 4]
```diff
@@ -56,30 +67,68 @@ var (
...
+		"timeout":                  1,
+		"unhealthy":                defaultActiveUnhealthy,
+	}
+	defaultPassive = map[string]interface{}{
+		"healthy":   defaultPassiveHealthy,
+		"type":      "http",
```
Comment (tjasko) [inline internal/resource/upstream.go:124]: nit:
```suggestion
		"type":      typedefs.ProtocolHTTP,
```

File: internal/resource/upstream.go [hunk 5]
```diff
@@ -122,6 +171,27 @@ var (
 			"grpc",
 			"grpcs",
 		},
+		Default: "http",
```
Comment (tjasko) [inline internal/resource/upstream.go:174]: nit: 
```suggestion
		Default: typedefs.ProtocolHTTP,
```

File: internal/model/json/validation/typedefs/fields.go [hunk 6]
```diff
@@ -16,6 +16,8 @@ const (
 	maxPathLength     = 1024
 	hostnamePattern   = "[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])?(.[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])?)*$"
```
Comment (tjasko) [inline internal/model/json/validation/typedefs/fields.go:18]: Should we write a unit test that creates each resource & asserts the non-zero values that were set to the relevant JSON schema? This way, we could have the test fail in the event a default value is missing on the JSON schema.

Comment (GGabriele) [reply internal/model/json/validation/typedefs/fields.go:18]: Right now we already have something like this checking if a resource with minimal configuration matches the default value: https://github.com/Kong/koko/blob/main/internal/resource/route_test.go#L38-L47

Do you think that's sufficient?

Comment (tjasko) [reply internal/model/json/validation/typedefs/fields.go:18]: I was actually talking about asserting that we're not missing any defaults on the JSON schema definition. What you're speaking of is a bit different & is simply just testing the defaults from the Kong Lua code are applied, not that we're setting the defaults on the JSON schema properly.

**Review by tjasko — APPROVED** [review]
The changes look good! While I still think this [comment](https://github.com/Kong/koko/pull/490#discussion_r1005989708) is something we should consider if it's actually possible, the work required isn't trivial & we should punt it off into another PR anyway... also that it speeds up getting this released anyway.

**Pull request merged**